PaintingTrivia.py

```diff
@@ -119,14 +128,49 @@ def show_entries():
             'selected_painter_id'] else 'wrong_guesses'
         session[key] += 1
 
-    painters = Painter.query.order_by(func.random()).limit(4).all()
+    while True:
+        painters = Painter.query.order_by(func.random()).limit(4).all()
+        selected_painter = random.choice(painters)
+        selected_painting = Painting.query.filter(Painting.painter == selected_painter).order_by(func.random()).limit(
+            1).first()
+        if selected_painting:
+            break
```

Comment:
smth along these lines: http://stackoverflow.com/questions/6044309/sqlalchemy-how-to-join-several-tables-by-one-query